### PR TITLE
Clean up user listening and auth.

### PR DESCRIPTION
**Pull request by WadePeterson**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -18,6 +18,7 @@
     }
   },
   "rules": {
+    "react/prop-types": "off",
```
Comment (WadePeterson) [inline .eslintrc]: Don't think this is super useful anyway, and there were so many errors. TypeScript props are superior to proptypes anyway, and they may come eventually!

File: src/app/utils/auth/AuthedRoute.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Redirect, Route, withRouter } from 'react-router-dom';
+
+const mapStateToProps = state => {
+  return { isAuthenticated: !!(state.currentUser && state.currentUser.uid) };
+};
+
+const AuthedRoute = (props) => {
```
Comment (WadePeterson) [inline src/app/utils/auth/AuthedRoute.js]: This is a React Router 4 way of handling routes that require auth.

File: .eslintrc (hunk 1 above)
Comment (mfernandez-rally) [inline .eslintrc]: Yay typescript!

**Comment by mfernandez-rally** [comment]
:egg: 🐰

**Pull request merged**